### PR TITLE
[DOCS] Clarifies data frame transform frequency is also for retries

### PR DESCRIPTION
**Pull request by droberts195**
In data frame transforms the same scheduler controls both
retries in the event of search failures and gaps between
checks for changes when the transform is running continuously.

I also standardized the wording between the two places where
`frequency` is documented.

### CODE REVIEW

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 1]
```diff
@@ -56,9 +56,11 @@ IMPORTANT:  You must use {kib} or this API to create a {dataframe-transform}.
   and optionally a `pipeline` id. See <<data-frame-transform-dest>>.
 
 `frequency`::
-  (time units) Optional. The interval between checks for changes in the source indices
-  when the {dataframe-transform} is running continuously. Defaults to `1m`.
-  The lowest permitted value is `1s`; the highest `1h`.
+  (time units) Optional. The interval between checks for changes in the source
+  indices when the {dataframe-transform} is running continuously. If there is
```
Comment (lcawl) [inline docs/reference/data-frames/apis/put-transform.asciidoc]: From discussions with @sophiec20 about the difference between data frame transforms with one or many checkpoints, I'm wondering whether this qualification is necessary.

```suggestion
  indices. If there is
```

**Review by hendrikmuhs — APPROVED** [review]
LGTM

File: docs/reference/data-frames/apis/put-transform.asciidoc (hunk 1 above)
Comment (sophiec20) [reply docs/reference/data-frames/apis/put-transform.asciidoc]: 1. `when .... is running continuously` -- yes, this is valid here.
2. `Also determines the retry interval in the event of transient failures during {dataframe-transform} search or indexing.` -- we recently added a retry to indexing, best to mention both

**Comment by droberts195** [comment]
@lcawl @sophiec20 I switched to using the wording of https://github.com/elastic/elasticsearch/pull/44229#discussion_r303561853 - are you happy to merge now?

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 2]
```diff
@@ -62,9 +62,11 @@ IMPORTANT:  You must use {kib} or this API to create a {dataframe-transform}.
     (Optional, string) The unique identifier for a <<pipeline,pipeline>>.
 
 `frequency`::
-  (Optional, time units) The interval between checks for changes in the source indices
-  when the {dataframe-transform} is running continuously. Defaults to `1m`.
-  The lowest permitted value is `1s`; the highest `1h`.
+  (Optional, time units) The interval between checks for changes in the source
+  indices when the {dataframe-transform} is running continuously. Also determines
+  the retry interval in the event of transient failures during {dataframe-transform}
```
Comment (lcawl) [inline docs/reference/data-frames/apis/put-transform.asciidoc]:
```suggestion
  the retry interval in the event of transient failures while the {dataframe-transform} is
```

File: docs/reference/data-frames/apis/put-transform.asciidoc [hunk 3]
```diff
@@ -62,9 +62,11 @@ IMPORTANT:  You must use {kib} or this API to create a {dataframe-transform}.
...
+  search or indexing. The minimum value is `1s` and the maximum is `1h`. The
```
Comment (lcawl) [inline docs/reference/data-frames/apis/put-transform.asciidoc]:
```suggestion
  searching or indexing. The minimum value is `1s` and the maximum is `1h`. The
```

File: docs/reference/data-frames/apis/transformresource.asciidoc [hunk 4]
```diff
@@ -21,8 +21,10 @@ For more information, see
   
 `frequency`::
   (time units) The interval between checks for changes in the source indices
-  when the {dataframe-transform} is running continuously. The minimum value is
-  `1s` and the maximum is `1h`. The default value is `1m`.
+  when the {dataframe-transform} is running continuously. Also determines the
+  retry interval in the event of transient failures during {dataframe-transform}
```
Comment (lcawl) [inline docs/reference/data-frames/apis/transformresource.asciidoc]:
```suggestion
  retry interval in the event of transient failures while the {dataframe-transform} is
```

File: docs/reference/data-frames/apis/transformresource.asciidoc [hunk 5]
```diff
@@ -21,8 +21,10 @@ For more information, see
...
+  search or indexing. The minimum value is `1s` and the maximum is `1h`. The
```
Comment (lcawl) [inline docs/reference/data-frames/apis/transformresource.asciidoc]:
```suggestion
  searching or indexing. The minimum value is `1s` and the maximum is `1h`. The
```

**Review by lcawl — APPROVED** [review]
Added minor comments, otherwise LGTM

**Comment by droberts195** [comment]
Thanks for the suggestions @lcawl.  I added them all.

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/docs

**Pull request merged**